crates/services/tx_status_manager/src/manager.rs

```diff
@@ -128,8 +149,19 @@ impl TxStatusManager {
 
     fn add_new_status(&mut self, tx_id: TxId, tx_status: TransactionStatus) {
         let now = Instant::now();
-        self.data.pruning_queue.push_front((now, tx_id));
-        self.data.statuses.insert(tx_id, (now, tx_status));
+        if TxStatusManager::is_prunable(&tx_status) {
+            self.data.pruning_queue.push_front((now, tx_id));
+            self.data.prunable_statuses.insert(tx_id, (now, tx_status));
+            self.data.non_prunable_statuses.remove(&tx_id);
+        } else {
+            if self.data.prunable_statuses.contains_key(&tx_id) {
+                // TODO[RC]: This should be shown in metrics: https://github.com/FuelLabs/fuel-core/issues/2851
+                tracing::warn!(%tx_id, "switching back to non-prunable status");
+                self.data.prunable_statuses.remove(&tx_id);
+                self.data.pruning_queue.retain(|(_, id)| id != &tx_id);
+            }
```

Comment:
```suggestion
```

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2853/commits/5bf74053b5a7d39b5daaa7231bf9f99c9a392d4d